mapper/orm/src/main/java/org/hibernate/search/mapper/orm/session/context/impl/HibernateOrmSessionContextImpl.java

```diff
@@ -21,6 +49,27 @@ public HibernateOrmSessionContextImpl(SessionImplementor sessionImplementor) {
 		this.runtimeIntrospector = new HibernateOrmRuntimeIntrospector( sessionImplementor );
 	}
 
+	@Override
+	@SuppressWarnings("unchecked")
+	public <T> T unwrap(Class<T> clazz) {
+		if ( clazz.isAssignableFrom( HibernateOrmSessionContext.class ) ) {
+			return (T) this;
+		}
+		/*
+		 * As a shortcut, we allow to unwrap directly to the Hibernate ORM session instead of
+		 * having to write .unwrap( HibernateOrmSessionContext.class ).getSession().
+		 */
+		else if ( SUPPORTED_SESSION_IMPLEMENTOR_UNWRAP_CLASSES.contains( clazz ) ) {
+			return (T) sessionImplementor;
+		}
```

Comment:
Not a big fan of it, to be honest.

It saves a method call but I don't think the additional code/complexity/potential lack of clarity is worth it.